packages/lib-user/src/components/MyGroups/components/GroupForm/GroupForm.js

```diff
@@ -0,0 +1,134 @@
+import { Box, Button, Form, FormField, RadioButtonGroup, Select, TextInput } from 'grommet'
+import { func, shape, string } from 'prop-types'
+import { useState } from 'react'
+import styled from 'styled-components'
+
+const StyledButton = styled(Button)`
+  border-radius: 4px;
+`
+
+const PRIVATE_STATS_VISIBILITY = [
+  {
+    label: 'No, never show individual stats',
+    value: 'private_agg_only',
+  },
+  {
+    label: 'Yes, always show individual stats',
+    value: 'private_show_agg_and_ind',
+  }
+]
+
+const PUBLIC_STATS_VISIBILITY = [
+  {
+    label: 'No, never show individual stats',
+    value: 'public_agg_only',
+  },
+  {
+    label: 'Yes, show individual stats if member',
+    value: 'public_agg_show_ind_if_member',
+  },
+  {
+    label: 'Yes, always show individual stats',
+    value: 'public_show_all',
+  }
+]
+
+const DEFAULT_HANDLER = () => true
+
+const DEFAULT_VALUE = {
+  display_name: '',
+  visibility: 'Private',
+  stats_visibility: 'private_agg_only'
+}
+
+function GroupForm({
+  handleSubmit = DEFAULT_HANDLER,
+  defaultValue = DEFAULT_VALUE
+}) {
+  const [value, setValue] = useState(defaultValue)
+  const statsVisibilityOptions = value.visibility === 'Private' ? PRIVATE_STATS_VISIBILITY : PUBLIC_STATS_VISIBILITY
+
+  return (
+    <Box
+      width={{ min: '600px' }}
+    >
+      <Form
+        onChange={(nextValue, { touched }) => {
+          if (nextValue.visibility !== value.visibility) {
+            const statsVisibility = nextValue.visibility === 'Private' ? 'private_agg_only' : 'public_agg_only'
+            nextValue.stats_visibility = statsVisibility
+          }
```

Comment:
This is setting the `stats_visibility` options per the Public or Private radio button group. The Public/Private field is not saved or part of a new user group, but used to simplify the `stats_visibility` options.